doc/source/serve/advanced.rst

```diff
@@ -85,28 +85,80 @@ dive.
+When calling ``set_traffic``, you provide a dictionary of backend name to a float value that will be used to randomly route that portion of traffic (out of a total of 1.0) to the given backend.
+For example, here we split traffic 50/50 between two backends:
 
 .. code-block:: python
-  
-  serve.create_endpoint("endpoint_identifier_split", "/split", methods=["GET", "POST"])
 
-  # splitting traffic 70/30
-  serve.set_traffic("endpoint_identifier_split", {"my_endpoint_backend": 0.7, "my_endpoint_backend_class": 0.3})
+  serve.create_endpoint("fifty-fifty", "/fifty")
+  serve.create_backend("backend1", MyClass1)
+  serve.create_backend("backend2", MyClass2)
+  serve.set_traffic("fifty-fifty", {"backend1": 0.5, "backend2": 0.5})
+
+Each request is routed randomly between the backends in the traffic dictionary according to the provided weights.
+Please see :ref:`session-affinity` for details on how to ensure that clients or users are consistently mapped to the same backend.
+
+A/B Testing
+~~~~~~~~~~~
+
+``set_traffic`` can be used to implement A/B testing by having one backend serve the majority of traffic while a fraction is routed to a second model:
+
+.. code-block:: python
+
+  serve.create_endpoint("ab_endpoint", "/a-b-test")
+  serve.create_backend("default_backend", MyClass)
+
+  # Initially, set all traffic to be served by the "default" backend.
+  serve.set_traffic("ab_endpoint", {"default_backend": 1.0})
+
+  # Add a second backend and route 1% of the traffic to it.
+  serve.create_backend("new_backend", MyNewClass)
+  serve.set_traffic("ab_endpoint", {"default_backend": 0.99, "new_backend": 0.01})
 
-While splitting traffic is general simple, at times you'll want to consider :ref:`session-affinity`, making it easy to
-control what users see which version of the model. See the docs on :ref:`session-affinity` for more information.
+  # Add a third backend that serves another 1% of the traffic.
+  serve.create_backend("new_backend2", MyNewClass2)
+  serve.set_traffic("ab_endpoint", {"default_backend": 0.98, "new_backend": 0.01, "new_backend2": 0.01})
+
+  # Revert to the "default" backend serving all traffic.
+  serve.set_traffic("ab_endpoint", {"default_backend": 1.0})
+
+Incremental Rollout
+~~~~~~~~~~~~~~~~~~~
+
+``set_traffic`` can also be used to implement incremental rollout.
+Here, we want to replace an existing backend with a new implementation by gradually increasing the proportion of traffic that it serves.
+
+.. code-block:: python
+
+  serve.create_endpoint("incremental_endpoint", "/incremental")
+  serve.create_backend("existing_backend", MyClass)
+
+  # Initially, all traffic is served by the existing backend.
+  serve.set_traffic("incremental_endpoint", {"existing_backend": 1.0})
+
+  # Then we can slowly increase the proportion of traffic served by the new backend.
+  serve.create_backend("new_backend", MyNewClass)
+  serve.set_traffic("incremental_endpoint", {"existing_backend": 0.9, "new_backend": 0.1})
```

Comment:
can you add a note that this can happen _over time_, not necessarily in one script?

Also, can we show an example of a rollback?